### ISSUE TITLE
Unclear errors when querying non-existent fields

### ISSUE THREAD

**Issue by alexkirsz** [dotansimha/graphql-code-generator#2515]
**Describe the bug**
Errors thrown by the typescript-operations plugin are extremely unclear. In this case, when querying a field that does not exist in the schema (`published` vs. `publishedAt`).

**To Reproduce**

1. My GraphQL schema:

I've removed as much noise as possible.

```graphql
"""
The `DateTime` scalar type represents a DateTime
value as specified by
[iso8601](https://en.wikipedia.org/wiki/ISO_8601).
"""
scalar DateTime

type InstanceNode implements Node {
  id: ID!
  publishedAt: DateTime
}

type InstanceNodeEdge {
  """The item at the end of the edge"""
  node: InstanceNode
}

type InstanceNodeConnection {
  edges: [InstanceNodeEdge]!
}

type Query {
  allInstances: InstanceNodeConnection!
}

```

2. My GraphQL operations:

```graphql
fragment InstanceCard on InstanceNode {
  published
}
```

```graphql
#import "./InstanceCard.gql"

query InstanceList {
  instances: allInstances {
    edges {
      node {
        id
        ...InstanceCard
      }
    }
  }
}

```

3. My `codegen.yml` config file:

I'm using the programmatic API, calling into `near-operation` preset's `buildGeneratesSection` method and then calling `codegen` on the configs it outputs.

I'm using `@graphql-codegen/typescript-operations` and `@graphql-codegen/typescript-react-apollo` with:
* `apolloReactHooksImportFrom` set to a custom path;
* `withHooks` set to `true`;
* `withComponent` set to `false`;
* `withHOC` set to `false`;
* `documentMode` set to `DocumentMode.graphQLTag`;
*  `preResolveTypes` set to `true`;
* `scalars` set to `{ DateTime: 'string', JSONString: 'string' }`;
* `namingConvention` set to `'keep'`.

**Current behavior**
`codegen` throws:

```
TypeError: Cannot read property 'type' of undefined
    at /Users/alexkirsz/project/node_modules/@graphql-codegen/visitor-plugin-common/src/selection-set-to-object.ts:478:45
    at Array.map (<anonymous>)
    at TypeScriptSelectionSetToObject.buildPrimitiveFieldsWithoutPick (/Users/alexkirsz/project/node_modules/@graphql-codegen/visitor-plugin-common/src/selection-set-to-object.ts:476:19)
    at TypeScriptSelectionSetToObject.buildSelectionSetString (/Users/alexkirsz/project/node_modules/@graphql-codegen/visitor-plugin-common/src/selection-set-to-object.ts:402:41)
    at /Users/alexkirsz/project/node_modules/@graphql-codegen/visitor-plugin-common/src/selection-set-to-object.ts:290:19
    at Array.map (<anonymous>)
    at TypeScriptSelectionSetToObject.get string [as string] (/Users/alexkirsz/project/node_modules/@graphql-codegen/visitor-plugin-common/src/selection-set-to-object.ts:283:43)
    at TypeScriptDocumentsVisitor.FragmentDefinition (/Users/alexkirsz/project/node_modules/@graphql-codegen/visitor-plugin-common/src/base-documents-visitor.ts:152:33)
    at Object.visit (/Users/alexkirsz/project/node_modules/graphql/language/visitor.js:242:26)
    at Object.exports.plugin (/Users/alexkirsz/project/node_modules/@graphql-codegen/typescript-operations/src/index.ts:78:25)
```

**Expected behavior**
Since I'm trying to query a field that does not exist on InstanceNode, I should be getting an error that tells me my queries do not match the schema, possibly with the location of the error.

**Environment:**

```json
{
    "@graphql-codegen/cli": "1.7.0",
    "@graphql-codegen/core": "1.7.0",
    "@graphql-codegen/near-operation-file-preset": "1.7.0",
    "@graphql-codegen/plugin-helpers": "1.7.0",
    "@graphql-codegen/typescript": "1.7.0",
    "@graphql-codegen/typescript-operations": "1.7.0",
    "@graphql-codegen/typescript-react-apollo": "1.7.0",
    "@graphql-codegen/visitor-plugin-common": "1.7.0"
}
```

- OS: macOS 10.14.6 (18G87)
- NodeJS: v12.10.0

**Additional context**

I'd also previously gotten a `TypeError: Invalid State`, although I can't remember exactly what was wrong with my code at the time. But the gist of the issue was the same: my queries did not properly match my schema.

**Comment by alexkirsz** [comment]
Is there perhaps a way to run a pre-validation step of my queries before feeding them to graphql-codegen?

**Comment by dotansimha** [comment]
Thanks @alexkirsz ! 
You are right, there should be a better error regarding this. I did some changes in `graphql-toolkit` recently that should fix the validation errors and catch them before. Any chance you can try the latest alpha and see if you are getting a better error? `1.7.1-alpha-8c70bf5d.24`

**Comment by alexkirsz** [comment]
Hey @dotansimha, I'm still getting the exact same error:

```
TypeError: Cannot read property 'type' of undefined
    at /Users/alex/project/node_modules/@graphql-codegen/visitor-plugin-common/src/selection-set-to-object.ts:478:45
    at Array.map (<anonymous>)
    at TypeScriptSelectionSetToObject.buildPrimitiveFieldsWithoutPick (/Users/alex/project/node_modules/@graphql-codegen/visitor-plugin-common/src/selection-set-to-object.ts:476:19)
    at TypeScriptSelectionSetToObject.buildSelectionSetString (/Users/alex/project/node_modules/@graphql-codegen/visitor-plugin-common/src/selection-set-to-object.ts:402:41)
    at /Users/alex/project/node_modules/@graphql-codegen/visitor-plugin-common/src/selection-set-to-object.ts:290:19
    at Array.map (<anonymous>)
    at TypeScriptSelectionSetToObject.get string [as string] (/Users/alex/project/node_modules/@graphql-codegen/visitor-plugin-common/src/selection-set-to-object.ts:283:43)
    at TypeScriptDocumentsVisitor.FragmentDefinition (/Users/alex/project/node_modules/@graphql-codegen/visitor-plugin-common/src/base-documents-visitor.ts:152:33)
    at Object.visit (/Users/alex/project/node_modules/graphql/language/visitor.js:242:26)
    at Object.exports.plugin (/Users/alex/project/node_modules/@graphql-codegen/typescript-operations/src/index.ts:78:25)
```

```json
{
    "@graphql-codegen/cli": "1.7.1-alpha-8c70bf5d.24",
    "@graphql-codegen/core": "1.7.1-alpha-8c70bf5d.24",
    "@graphql-codegen/near-operation-file-preset": "1.7.1-alpha-8c70bf5d.24",
    "@graphql-codegen/plugin-helpers": "1.7.1-alpha-8c70bf5d.24",
    "@graphql-codegen/typescript": "1.7.1-alpha-8c70bf5d.24",
    "@graphql-codegen/typescript-graphql-files-modules": "1.7.1-alpha-8c70bf5d.24",
    "@graphql-codegen/typescript-operations": "1.7.1-alpha-8c70bf5d.24",
    "@graphql-codegen/typescript-react-apollo": "1.7.1-alpha-8c70bf5d.24",
    "@graphql-codegen/visitor-plugin-common": "1.7.1-alpha-8c70bf5d.24"
}
```

**Comment by dotansimha** [comment]
I think I found the issues, it caused by `near-operation-file`, because we allow to skip validation there. I'll try to fix that.

**Comment by dotansimha** [comment]
@alexkirsz 
I think I fixed it in: https://github.com/dotansimha/graphql-code-generator/pull/2531 
Can you please try `1.7.1-alpha-7be0e2a0.33`?

**Comment by dotansimha** [comment]
Fixed in 1.8.0 🚀

**Comment by alexkirsz** [comment]
@dotansimha I'm now getting an unclear error on 1.8.0:

```
TypeError: Cannot read property 'getType' of undefined
    at /project/node_modules/@graphql-codegen/near-operation-file-preset/src/index.ts:154:50
    at Array.reduce (<anonymous>)
    at Object.buildGeneratesSection (/project/node_modules/@graphql-codegen/near-operation-file-preset/src/index.ts:148:70)
    at generate (/project/scripts/generateTypes.ts:392:47)
    at main (/project/scripts/generateTypes.ts:470:5)
```

On 1.7.0, I was getting the following instead:
```
TypeError: Could not find field type. InstanceNode.subject
    at TypeScriptSelectionSetToObject.buildSelectionSetString (/project/node_modules/@graphql-codegen/visitor-plugin-common/src/selection-set-to-object.ts:360:19)
    at /project/node_modules/@graphql-codegen/visitor-plugin-common/src/selection-set-to-object.ts:290:19
    at Array.map (<anonymous>)
    at TypeScriptSelectionSetToObject.get string [as string] (/project/node_modules/@graphql-codegen/visitor-plugin-common/src/selection-set-to-object.ts:283:43)
    at TypeScriptSelectionSetToObject.buildSelectionSetString (/project/node_modules/@graphql-codegen/visitor-plugin-common/src/selection-set-to-object.ts:389:63)
    at /project/node_modules/@graphql-codegen/visitor-plugin-common/src/selection-set-to-object.ts:290:19
    at Array.map (<anonymous>)
    at TypeScriptSelectionSetToObject.get string [as string] (/project/node_modules/@graphql-codegen/visitor-plugin-common/src/selection-set-to-object.ts:283:43)
    at TypeScriptDocumentsVisitor.OperationDefinition (/project/node_modules/@graphql-codegen/visitor-plugin-common/src/base-documents-visitor.ts:175:33)
    at Object.visit (/project/node_modules/graphql/language/visitor.js:242:26)
```

**Comment by dotansimha** [comment]
Thanks for the feedback @alexkirsz , I'll take a look.

**Comment by dotansimha** [comment]
@alexkirsz I think I managed to fix that in https://github.com/dotansimha/graphql-code-generator/pull/2756 

Also, tried with the examples with have in the repo, and it seems like it works, I'm getting the correct validation error instead of the error you mentioned.

Can you please try `1.8.2-alpha-a98ee5eb.42`? And if it doesn't work for you, can you please share a reproduction in a repo/sandbox? 

Thanks!

**Comment by dotansimha** [comment]
Fixed in 1.8.2.
